src/helpers/binding.ts

```diff
@@ -0,0 +1,163 @@
+const placeholderRegExp =
+  /["'`](?:(?<=")[^"\\]*(?:\\.[^"\\]*)*"|(?<=')[^'\\]*(?:\\.[^'\\]*)*'|[^`]*`)|(\?\d*|[:@$][a-zA-Z_]\w+)/gmu;
+
+function isPlainObject(obj: any): obj is Record<string, BaseType> {
+  if (typeof obj !== "object" || obj === null) return false;
+  const proto = Object.getPrototypeOf(obj);
+  return proto !== null && Object.getPrototypeOf(proto) === null;
+}
+
+function bindString(param: string, quoteEscaper = "''"): string {
+  return "'" + param.replace(/'/g, quoteEscaper) + "'";
+}
+
+function bindBoolean(param: boolean): string {
+  return Number(param).toString();
+}
+
+function bindNumber(param: number | bigint): string {
+  return param.toString();
+}
+
+function bindDate(param: Date): string {
+  return param.valueOf().toString();
+}
+
+function bindBytes(param: Uint8Array): string {
+  const hex = param.reduce((t, x) => t + x.toString(16).padStart(2, "0"), "");
+  return `X'${hex}'`;
+}
+
+function bindObject(param: any): string {
+  return bindString(JSON.stringify(param));
+}
+
+function bindNull(_param: undefined | null): string {
+  return "NULL";
+}
+
+function bindToString(param: any): string {
+  return bindString(String(param));
+}
+
+interface SQL {
```

Comment:
A little bit of a magic thing if someone wants to stick a custom object into a table. This will likely be useful for things like auto-pinning data to IPFS and including the hash and things like that. This way, we have a single interface that the lib knows about to do all sorts of custom things.